src/NHapi.Base/Model/Primitive/TSComponentOne.cs

```diff
@@ -24,309 +24,342 @@
+        }
+
+        /// <seealso cref="CommonTS.setDateMinutePrecision(int, int, int, int, int)">
+        /// </seealso>
+        /// <throws>  DataTypeException if the value is incorrectly formatted.  If validation is enabled, this.  </throws>
+        /// <summary>      exception should be thrown at setValue(), but if not, detailed parsing may be deferred until
+        /// this method is called.
+        /// </summary>
+        public virtual void SetDateMinutePrecision(int yr, int mnth, int dy, int hr, int min)
+        {
+            Detail.SetDateMinutePrecision(yr, mnth, dy, hr, min);
+        }
+
+        [Obsolete("This method has been replaced by 'SetDateSecondPrecision'.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "StyleCop.CSharp.NamingRules",
+            "SA1300:Element should begin with upper-case letter",
+            Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+        public virtual void setDateSecondPrecision(int yr, int mnth, int dy, int hr, int min, float sec)
+        {
+            SetDateSecondPrecision(yr, mnth, dy, hr, min, sec);
+        }
+
+        /// <seealso cref="CommonTS.setDateSecondPrecision(int, int, int, int, int, float)">
+        /// </seealso>
+        /// <throws>  DataTypeException if the value is incorrectly formatted.  If validation is enabled, this.  </throws>
+        /// <summary>      exception should be thrown at setValue(), but if not, detailed parsing may be deferred until
+        /// this method is called.
+        /// </summary>
+        public virtual void SetDateSecondPrecision(int yr, int mnth, int dy, int hr, int min, float sec)
+        {
+            Detail.SetDateSecondPrecision(yr, mnth, dy, hr, min, sec);
+        }
+
+        /// <summary>
+        /// Get the value as a date.  Throws hl7Exception if error.
+        /// </summary>
+        /// <returns>Data/Time.</returns>
+        public virtual DateTime GetAsDate()
+        {
+            try
+            {
+                var dateFormats = new string[] { LongDateTimeFormat, ShortDateTimeFormat, LongDateTimeFormatWithSecond, LongDateTimeFormatWithOffset, LongDateTimeFormatWithFractionOfSecond };
+                var val = DateTime.MinValue;
+                var culture = Thread.CurrentThread.CurrentCulture;
+                if (Value != null && Value.Length > 0)
+                {
+                    val = DateTime.ParseExact(Value, dateFormats, culture, DateTimeStyles.NoCurrentDateDefault);
+                }
+
+                return val;
+            }
+            catch (Exception)
+            {
+                throw new HL7Exception("Could not get field as dateTime");
+            }
+        }
+
+        /// <summary>
+        /// Set the value as a short date.
+        /// </summary>
+        /// <param name="value"></param>
+        public virtual void SetShortDate(DateTime value)

```

Comment:
these methods could be expression bodies too (if we like that sort of thing, check the stylecop rules)

Reply:
personally, I don't like expression body methods - I think it makes things _harder_ to read, not easier. Also makes it easier to distiguish (at a glance) between methods and properties. Thoughts?

Reply:
I think I agree 👍🏼